packages/eas-cli/src/utils/profiles.ts

```diff
@@ -47,16 +54,53 @@ export async function getProfilesAsync<T extends ProfileType>({
   return await Promise.all(results);
 }
 
-async function readProfileAsync<T extends ProfileType>({
+async function setNodeVersionFromFileAsync(
+  projectDir: string,
+  profile: BuildProfile<Platform>
+): Promise<void> {
+  if (profile?.node) {
+    return;
+  }
```

Comment:
```suggestion
async function maybeSetNodeVersionFromFileAsync(
  projectDir: string,
  profile: BuildProfile<Platform>
): Promise<void> {
  if (profile?.node) {
    return;
  }
```
If `profile?.node` is set then it is not set from file